### ISSUE TITLE
include dalvik.gdb in NDK (and use it in ndk-gdb)

### ISSUE THREAD

**Issue by rprichard** [android-ndk/ndk#637]
Refiled from internal issue http://b/23715351:

without this, debugging stuff with art is a PITA:

https://android.googlesource.com/platform/development/+/73cd247e450dc8732a22dbd4bff95c18bfc968ac/scripts/gdb/dalvik.gdb#16

**Comment by DanAlbert** [comment]
Actually this should be really simple and doable in r17, right?

**Comment by rprichard** [comment]
Including dalvik.gdb is easy.

Something needs to be done with ndk-gdb, because that script assumes the pre-Studio Android project + ndk-build. It doesn't work with either CMake and/or Gradle projects.

**Comment by DanAlbert** [comment]
Yeah, worth filing a bug to track the latter, but we can at least get dalvik.gdb into r17.

**Comment by rprichard** [comment]
For dalvik.gdb's `art-on` to work, we need to pull `libart.so` from the device, but it won't be there on Jelly Bean, so I'll check whether it exists before pulling it.

The existence of `/system/lib[64]/libart.so` doesn't imply the use of ART (instead of Dalvik) on older devices, so I suppose we'll need to detect whether ART is in use. Maybe something like `if &art_sigsegv_fault != 0`?

Apparently the `art_sigsegv_fault` breakpoint method isn't reliable until P because it can be inlined: (http://b/77528455, and maybe http://b/77492857 too)

**Comment by enh** [comment]
maybe until we can do all the support work, we should just have some docs somewhere pointing to the platform dalvik.gdb and recommending it?
